### PR TITLE
ARTEMIS-2085 - Improve validation of MDB activation config properties values

### PR DESCRIPTION
**Pull request by rpelisse**
Issue: https://issues.jboss.org/browse/JBEAP-14106
No upstream issue
Upstream PR: https://github.com/apache/activemq-artemis/pull/2312

### CODE REVIEW

**Comment by michaelandrepearce** [comment]
Please can you create an activemq jira and update all references to that,not to a commercial vendors internal bugs system

File: artemis-ra/src/main/java/org/apache/activemq/artemis/ra/inflow/ActiveMQActivation.java [hunk 1]
```diff
@@ -556,7 +556,7 @@ protected void setupDestination() throws Exception {
                   calculatedDestinationName = spec.getQueuePrefix() + calculatedDestinationName;
                }
 
-               logger.debug("Unable to retrieve " + destinationName + " from JNDI. Creating a new " + destinationType.getName() + " named " + calculatedDestinationName + " to be used by the MDB.");
+               logger.warn("Unable to retrieve " + destinationName + " from JNDI. Creating a new " + destinationType.getName() + " named " + calculatedDestinationName + " to be used by the MDB.");
```
Comment (michaelandrepearce) [inline artemis-ra/src/main/java/org/apache/activemq/artemis/ra/inflow/ActiveMQActivation.java]: As a warn this should use a registered logger with a logger code

File: artemis-ra/src/main/java/org/apache/activemq/artemis/ra/inflow/ActiveMQActivationSpec.java [hunk 2]
```diff
@@ -406,7 +406,9 @@ public void setAcknowledgeMode(final String value) {
       } else if ("AUTO_ACKNOWLEDGE".equalsIgnoreCase(value) || "Auto-acknowledge".equalsIgnoreCase(value)) {
          acknowledgeMode = Session.AUTO_ACKNOWLEDGE;
       } else {
-         throw new IllegalArgumentException("Unsupported acknowledgement mode " + value);
+         final String message = "Unsupported acknowledgement mode " + value;
+         logger.warn(message);
```
Comment (michaelandrepearce) [inline artemis-ra/src/main/java/org/apache/activemq/artemis/ra/inflow/ActiveMQActivationSpec.java]: As warn requires a registered logger and logger code

File: artemis-ra/src/main/java/org/apache/activemq/artemis/ra/inflow/ActiveMQActivationSpec.java [hunk 3]
```diff
@@ -603,7 +605,11 @@ public void setMaxSession(final Integer value) {
          logger.trace("setMaxSession(" + value + ")");
       }
 
-      maxSession = value;
+      if ( value < 0 ) {
+         logger.warn("Invalid number of session (negative):" + value +", defaulting to 1.");
```
Comment (michaelandrepearce) [inline artemis-ra/src/main/java/org/apache/activemq/artemis/ra/inflow/ActiveMQActivationSpec.java]: Ditto

**Comment by clebertsuconic** [comment]
close this one please...


We merge into master.. than cherry-pick -x into 2.6.x

**Comment by rpelisse** [comment]
OK! I'll re open it when it is merge in master.

**Comment by clebertsuconic** [comment]
No need. I will cherry pick it myself. 


Unless there are conflicts.  But I do not think it’s the case.

**Pull request closed without merging**